README.md

```diff
@@ -221,3 +224,72 @@ Not all steps need to be followed on the instructions for the aad-pod-identity p
     ```
 
 **NOTE** When using the `Pod Identity` option mode, there can be some amount of delay in obtaining the objects from keyvault. During the pod creation time, in this particular mode `aad-pod-identity` will need to create the `AzureAssignedIdentity` for the pod based on the `AzureIdentity` and `AzureIdentityBinding`, retrieve token for keyvault. This process can take time to complete and it's possible for the pod volume mount to fail during this time. When the volume mount fails, kubelet will keep retrying until it succeeds. So the volume mount will eventually succeed after the whole process for retrieving the token is complete.
+
+
+## Local End-To-End Testing for the Key Vault Azure Provider
+
+This section will show you how to locally test the Azure Key Vault Provider end-to-end (e2e). The e2e tests utilize Bats for testing the scripts. Take a look inside the `test/bats/tests/local` folder to see the tests and the deployments needed for creating the e2e tests.
+
+### Prerequisites
+- [ Helm >= 3](https://helm.sh/)
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [Azure Service Principal](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest)
+- [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-manage-with-cli2)
+- [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/)
+- [Docker](https://docs.docker.com/get-started/)
+- [Bats](https://github.com/bats-core/bats-core)
+
+
```

Comment:
Where is the step to create Azure Key Vault and add the secrets?

Reply:
The `Azure Key Vault` bullet point has the link to key vault creation and secret setting inside of this article

Reply:
I think it needs to be explicitly called out as you will need to add a specific set of secrets to ensure the e2e tests succeed right?

Reply:
I set up the `local-e2e-test` to run with whatever objects they decide to provide to their Key Vault